### PR TITLE
feat(eval): supplement eval framework with CI gate, retrieval metrics, adversarial scenarios, and cost tracking

### PR DESCRIPTION
**Pull request by norrietaylor**
## Summary

Supplements the existing 56-scenario eval framework with four improvements:

- **promptfoo PR CI Gate**: Fast pre-merge eval gate with 10 smoke-test scenarios covering critical skill paths (`/distill`, `/recall`, `/bookmark`, `/pour`, `/watch`). Runs via `eval-pr.yml` on every PR targeting main, using promptfoo's native MCP stdio provider.
- **RAGAS Retrieval Quality Metrics**: `retrieval_scorer.py` computing precision@k, recall@k, MRR with threshold enforcement (MRR ≥ 0.7, precision@5 ≥ 0.6). Golden dataset with 5 retrieval scenarios. Graceful fallback when RAGAS not installed.
- **Adversarial & Edge-Case Scenarios**: 12 scenarios across 5 categories — malformed input, empty store, boundary conditions (dedup thresholds), concurrent operations, and missing dependencies. All assert no stack traces and user-friendly messaging.
- **Cost Tracking & Trend Detection**: Extended baseline JSON with `cost_summary` (per-skill token/cost breakdown). New `--compare-cost` flag detects >20% per-skill cost regressions. Backward compatible with old baseline format.

## Key files

| Area | Files |
|------|-------|
| promptfoo | `promptfooconfig.yaml`, `.github/workflows/eval-pr.yml` |
| Retrieval | `src/distillery/eval/retrieval_scorer.py`, `tests/eval/golden/retrieval.yaml` |
| Adversarial | `tests/eval/scenarios/adversarial.yaml` |
| Cost tracking | `src/distillery/cli.py` (extended) |
| Tests | `tests/eval/test_retrieval_scorer.py` (41 tests), `tests/eval/test_cost_tracking.py` (20 tests) |

## Test plan

- [x] `pytest -m unit tests/eval/test_retrieval_scorer.py` — 41/41 passing
- [x] `pytest -m unit tests/eval/test_cost_tracking.py` — 20/20 passing
- [x] `mypy --strict src/distillery/` — clean
- [x] `ruff check src/ tests/` — clean
- [x] Adversarial scenarios load via `load_scenarios_from_dir()` (12 scenarios)
- [x] `npx promptfoo@latest validate config --config promptfooconfig.yaml` — valid
- [ ] PR triggers `eval-pr.yml` workflow and passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * PR CI eval gate with prompt-based smoke tests and non-blocking model-run reporting; nightly eval picks up optional retrieval extras.
  * Retrieval-quality metrics (precision, recall, MRR, optional faithfulness) surfaced in eval outputs and can mark failing scenarios.
  * Cost tracking and compare mode with per-skill deltas and warnings for >20% increases.

* **Documentation**
  * New evaluation spec supplement, adversarial/edge-case, PR-gate, ragas-retrieval, and cost-tracking specs.

* **Tests**
  * Extensive tests for retrieval scoring and cost-tracking/comparison behavior.

* **Chores**
  * Ignore local prompt evaluation output directory.

### CODE REVIEW

File: docs/specs/02-spec-eval-supplement/02-proofs/T02.3-proofs.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+**Threshold constants**: `_MRR_THRESHOLD = 0.7`, `_PRECISION_THRESHOLD = 0.6`, and
+`_RETRIEVAL_K = 5` are defined as module-level constants for easy tuning.
+
+## Verification
+
+```
+ruff check src/distillery/cli.py    → All checks passed
+mypy --strict src/distillery/cli.py → Success: no issues found
+pytest -m unit -q                   → 875 passed, 2 skipped
+```
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply docs/specs/02-spec-eval-supplement/02-proofs/T02.3-proofs.md]: This file has been removed from the PR — proof artifacts are now gitignored and no longer tracked.

File: docs/specs/02-spec-eval-supplement/02-spec-eval-supplement.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+2. Measure retrieval relevance (precision, recall, MRR) for search-dependent skills using RAGAS
+3. Cover failure modes — malformed input, empty stores, boundary conditions, concurrency, dependency failures — with ≥10 adversarial scenarios
+4. Track token usage and estimated cost per eval run with baseline comparison and per-skill breakdown
+
+## User Stories
+
+- As a **contributor**, I want a fast eval check on my PR so that I know if my changes break skill behavior before merging.
+- As a **maintainer**, I want retrieval quality metrics so that I can detect degradation in search relevance beyond "correct tools were called."
+- As a **developer**, I want adversarial test scenarios so that edge cases and failure modes are covered by the eval suite.
+- As a **operator**, I want cost tracking per eval run so that I can detect prompt bloat or unnecessary tool calls driving up spend.
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply docs/specs/02-spec-eval-supplement/02-spec-eval-supplement.md]: Fixed in e1a15cb.

File: src/distillery/cli.py [hunk 3]
```diff
@@ -514,54 +702,113 @@ async def _run_all() -> list[Any]:
...
-                    "tool_call_count": r.performance.tool_call_count,
-                    "tools_called": r.effectiveness.tools_called,
-                    "failure_reasons": r.effectiveness.failure_reasons,
-                }
-            )
-        summary = {"results": output, "passed": passed, "total": total, "pass_rate": pass_rate}
-        print(json.dumps(summary, indent=2))
+        # Defer printing when compare_cost is active so we can append cost_comparison.
+        if not compare_cost:
+            print(json.dumps(summary, indent=2))
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply src/distillery/cli.py]: Fixed in e1a15cb — deferred JSON summary is now always printed when `--compare-cost` is active, even without a valid baseline.

File: docs/specs/02-spec-eval-supplement/02-spec-eval-supplement.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+| Retrieval MRR on `/recall` scenarios | ≥ 0.7 |
+| Retrieval precision@5 on `/recall` scenarios | ≥ 0.6 |
+| Adversarial scenario count | ≥ 10 |
+| Adversarial scenario pass rate | 100% (graceful handling) |
+| Cost tracking coverage | All scenarios in baseline JSON |
+| Cost regression detection threshold | >20% per-skill increase flagged |
+
+## Open Questions
+
+1. **promptfoo MCP provider**: Does promptfoo natively support MCP tool-call providers, or will a custom provider wrapper be needed to bridge promptfoo → Distillery MCP? This affects Unit 1 implementation complexity.
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply docs/specs/02-spec-eval-supplement/02-spec-eval-supplement.md]: Fixed in e1a15cb — all three questions resolved with implementation decisions.

File: docs/specs/02-spec-eval-supplement/promptfoo-pr-ci-gate.feature [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# Recommended test type: Integration
+
+Feature: promptfoo PR CI Gate
+
+  Scenario: promptfoo config defines smoke-test scenarios for critical skill paths
+    Given the Distillery repository is checked out
+    When the user runs "npx promptfoo@latest eval --config promptfooconfig.yaml --dry-run"
+    Then the command exits with code 0
+    And stdout lists at least 5 test scenarios
+    And the listed scenarios cover distill, recall, bookmark, pour, and watch skills
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply docs/specs/02-spec-eval-supplement/promptfoo-pr-ci-gate.feature]: The `.feature` files are informational Gherkin specs generated by the planning tool — they're not enforced in CI. The actual scenario count is validated by the promptfoo config validation step. Won't fix in this PR.

File: tests/eval/scenarios/adversarial.yaml [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+    - distillery_check_dedup
+  response_must_contain:
+    - already
+  response_must_not_contain:
+    - Traceback
+    - Exception
+    - internal error
+    - "Error:"
+  max_latency_ms: 90000
+  max_total_tokens: 10000
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply tests/eval/scenarios/adversarial.yaml]: Acknowledged — boundary threshold scenarios depend on embedding similarity which varies with model changes. These scenarios are designed for the nightly eval (not PR gate) where flakiness is acceptable as a signal. Adding deterministic dedup-threshold unit tests is tracked separately.
